networkx/algorithms/tree/branchings.py

```diff
@@ -363,150 +361,149 @@ def desired_edge(v):
 
             return edge, weight
 
-        while True:
-            # (I1): Choose a node v in G^i not in D^i.
-            try:
-               v = next(nodes)
-            except StopIteration:
-                # If there are no more new nodes to consider, then we *should*
-                # meet the break condition (b) from the paper:
-                #   (b) every node of G^i is in D^i and E^i is a branching
-                # Construction guarantees that it's a branching.
-                assert( len(G) == len(B) )
-                if len(B):
-                    assert( is_branching(B) )
-
-                if self.store:
-                    self.graphs.append(G.copy())
-                    self.branchings.append(B.copy())
+        def get_final_graph_and_branch(nodes):
+            D = set()
+            while True:
+                # (I1): Choose a node v in G^i not in D^i.
+                try:
+                   v = next(nodes)
+                except StopIteration:
+                    # If there are no more new nodes to consider, then we *should*
+                    # meet the break condition (b) from the paper:
+                    #   (b) every node of G^i is in D^i and E^i is a branching
+                    # Construction guarantees that it's a branching.
+                    if len(G) != len(B):
+                        raise Exception('Graph and branching must be of the same lenght.')
```

Comment:
The word is "length". However, a better message would be "graph and branching must have the same number of nodes", since a graph doesn't really have a "length".